### PR TITLE
Validate the pattern of sourceId

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?
So we don't have to write a in code validator in dor-services-app and we don't make invalid requests.

## How was this change tested?
Test suite.

## Which documentation and/or configurations were updated?
n/a

### CODE REVIEW

**Comment by justinlittman** [comment]
* Why is this pattern required?
* What are the parts of the pattern?

**Comment by jcoyne** [comment]
@justinlittman I don't know why this validation is necessary, but it's the convention we currently enforce. We currently validate this at the Fedora 3 model level here: https://github.com/sul-dlss/dor-services/blob/0e35e78e41125c2269a381d7c8d131bc9e766867/lib/dor/datastreams/identity_metadata_ds.rb#L63-L65  so it makes sense to transfer this into the cocina-model.

**Comment by justinlittman** [comment]
OK, then we should document in the openapi that the pattern should be `namespace:value`.

File: openapi.yml [hunk 1]
```diff
@@ -977,12 +978,13 @@ components:
       properties:
         sourceId:
```
Comment (mjgiarlo) [inline openapi.yml:979]: I wonder if this repetition means we ought to extract source id as its own component which can then be referred to via $ref wherever needed?

Comment (jcoyne) [reply openapi.yml:970]: done

**Pull request merged**